gnat2goto/driver/tree_walk.adb

```diff
@@ -2343,63 +2343,219 @@ package body Tree_Walk is
+            --  considered for service. This may change the behaviour.
+            Report_Unhandled_Node_Empty (N, "Do_Pragma",
+                                         "Unsupported pragma: Queuing Policy");
+         when Name_Remote_Types =>
+            --  Defines types intended for use in communication between active
+            --  partitions. We do not support concurrency.
+            Report_Unhandled_Node_Empty (N, "Do_Pragma",
+                                         "Unsupported pragma: Remote Types");
+         when Name_Restrictions =>
+            --  Expresses the user's intent to abide by certain restrictions.
+            --  This could probably be implemented as an assertion eventually.
+            Report_Unhandled_Node_Empty (N, "Do_Pragma",
+                                         "Unsupported pragma: Restrictions");
+         when Name_Shared_Passive =>
+            --  Used for managing global data shared between active partitions.
+            --  We do not support concurrency.
+            Report_Unhandled_Node_Empty (N, "Do_Pragma",
+                                         "Unsupported pragma: Shared Passive");
+         when Name_Task_Dispatching_Policy =>
+            --  Specifies the details of task dispatching that are not covered
+            --  by the basic task dispatching model. We do not support
+            --  concurrency.
+            Report_Unhandled_Node_Empty (N, "Do_Pragma",
+                                "Unsupported pragma: Task Dispatching Policy");
          when Name_All_Calls_Remote |
-              Name_Asynchronous |
-              Name_Atomic |
-              Name_Atomic_Components |
-              Name_Attach_Handler |
+              Name_Remote_Call_Interface |
+            --  Library unit pragma; used by the distributed systems annex
+            --  Interface for remote function calls between active partitions
+            --  Should not alter the semantics -> Ignored
+              Name_Annotate |
+            --  Ignore here. Rather look for those when we process a node.
+              Name_Assertion_Policy |
+            --  Control the pragma Assert according to the policy identifier
+            --  which can be Check, Ignore, or implementation-defined.
+            --  Ignore means that assertions are ignored at run-time -> Ignored
+              Name_Interrupt_Handler |
+            --  If the pragma appears in a protected_definition, then the
+            --  corresponding procedure can be attached ynamically, as a
+            --  handler, to interrupts. -> Ignored
               Name_Controlled |
-              Name_Convention |
-              Name_Discard_Names |
-              Name_Elaborate |
-              Name_Elaborate_All |
-              Name_Elaborate_Body |
+            --  Used to prevent any automatic reclamation of storage (garbage
+            --  collection) for the objects created by allocators of a given
+            --  access type. -> Ignored
```

Comment:
We have to consider resource allocation issues, so this needs to be "known but unsupported".

Reply:
Ok.